pkg/controller/namespace/namespace_controller_utils.go

```diff
@@ -244,6 +244,9 @@ func syncNamespace(kubeClient clientset.Interface, versions *unversioned.APIVers
 	// we have removed content, so mark it finalized by us
 	result, err := retryOnConflictError(kubeClient, namespace, finalizeNamespaceFunc)
 	if err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
It's not possible for a namespace to be deleted when its list of finalizers is non-empty.

see: https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/namespace/etcd/etcd.go#L110

So if we ever got a not found error here, the namespace is gone, and all content should be gone.  I worried about the error here in cases where a cluster found itself running two instances of controllermanager inadvertently.